packages/esm-esbuild/esbuild.js

```diff
@@ -0,0 +1,21 @@
+import { build } from "esbuild";
+
+build({
+  entryPoints: ["./src/index.ts"],
+  outdir: "./dist",
+  bundle: true,
+  // platform: "node",
+  format: "esm",
+  // target: "node16",
+  plugins: [
+    {
+      name: "add-mjs",
+      setup(build) {
+        build.onResolve({ filter: /.*/ }, (args) => {
```

Comment:
I think it could be made a bit smarter - f.i. check if `.mjs` file exists and only then include it, but I feel this become even more hackier. I'm not really understand why esbuild can't find relevant file by itself: afaik it includes some heuristics to resolve relevant imports, following the rules of Node itself.